ert3/storage/_storage.py

```diff
@@ -28,16 +30,98 @@ class Config:
     record_type: ert3.data.RecordType
 
 
+def _assert_server_info() -> None:
+    global _STORAGE_URL, _STORAGE_TOKEN  # pylint: disable=global-statement
+
+    if _STORAGE_URL is None:
+        info = get_info()
+        _STORAGE_URL = info["baseurl"]
+        _STORAGE_TOKEN = info["auth"][1]
+
+
+def _get_from_server(
+    url: str,
+    headers: Optional[Dict[Any, Any]] = None,
+    status_code: int = 200,
+    **kwargs: Dict[Any, Any],
+) -> requests.Response:
+
+    _assert_server_info()
+    if not headers:
+        headers = {}
+    headers["Token"] = _STORAGE_TOKEN
+
+    resp = requests.get(f"{_STORAGE_URL}/{url}", headers=headers, **kwargs)
+    if resp.status_code != status_code:
```

Comment:
Could you motivate why we are logging for non-successful requests here, while not raising an exception until we reach the caller? Should not the responsibility of first line error management be co-located?

Reply:
this is merely a helper function for getting, similar to posting and deleting. The handling of the return code is done the same as before like in line 328. I was aiming for minimal invasive change in the code. Thus at the moment keeping the error in the log for now.

Reply:
Continued in https://github.com/equinor/ert/issues/1735